htmd/parameterization/dihedral.py

```diff
@@ -366,16 +328,15 @@ def _fit(self):
         logger.info('Finished parameter optimization')
 
         # Update the target dihedral with the optimized parameters
-        self.parameters = self._vectorToParams(vector, self.parameters)
-        self.molecule._prm.updateDihedral(self.parameters)
+        self._updateDihedrals(self.parameters, self.fitdihkeys, vector)
```

Comment:
This should return an updated copy:
```python
self.parameters = self._vectorToParams(vector, self.parameters, self.fitdihkeys)
```

Reply:
I don't use `vectorToParams` anymore. I use directly the `_updateDihedrals` method.